lib/modules/datasource/go/releases-goproxy.ts

```diff
@@ -52,21 +60,31 @@ export class GoProxyDatasource extends Datasource {
           break;
         }
 
-        const releasesIndex = await this.listVersions(url, packageName);
-        const releases = await p.map(releasesIndex, async (versionInfo) => {
-          const { version, releaseTimestamp } = versionInfo;
-
-          if (releaseTimestamp) {
-            return { version, releaseTimestamp };
-          }
-
+        let releases: Release[] = [];
+        for (let i = 0; ; i++) {
           try {
-            return await this.versionInfo(url, packageName, version);
+            const major = currentMajor + i;
+            let mod: string;
+            if (major < 2 && !isGopkgin) {
+              mod = baseMod;
+              i++; // v0 and v1 are the same module
+            } else {
+              mod = `${baseMod}${majorSuffixSeparator}v${major}`;
+            }
+            const result = await this.getVersionsWithInfo(url, mod);
+            if (result.length === 0) {
+              break;
+            }
+            releases = releases.concat(result);
```

Comment:
```suggestion
            releases.push(...result);
```
https://dev.to/uilicious/javascript-array-push-is-945x-faster-than-array-concat-1oki